lib/ansible/modules/cloud/amazon/iam_server_certificate_facts.py

```diff
@@ -81,89 +89,88 @@
+    """
+    Get one or more IAM server certificates facts based on name or path. If not found, return None.
 
-def get_server_certs(iam, name=None):
-    """Retrieve the attributes of a server certificate if it exists or all certs.
-    Args:
-        iam (botocore.client.IAM): The boto3 iam instance.
-
-    Kwargs:
-        name (str): The name of the server certificate.
-
-    Basic Usage:
-        >>> import boto3
-        >>> iam = boto3.client('iam')
-        >>> name = "server-cert-name"
-        >>> results = get_server_certs(iam, name)
-        {
-            "upload_date": "2015-04-25T00:36:40+00:00",
-            "server_certificate_id": "ADWAJXWTZAXIPIMQHMJPO",
-            "certificate_body": "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----",
-            "server_certificate_name": "server-cert-name",
-            "expiration": "2017-06-15T12:00:00+00:00",
-            "path": "/",
-            "arn": "arn:aws:iam::911277865346:server-certificate/server-cert-name"
-        }
+    :param connection: AWS boto3 glue connection
+    :param module: Ansible module
+    :param name: The name of the server certificate
+    :param path: The path of the server certificate
+    :return: IAM server certificates dict or None if not found
     """
-    results = dict()
+
+    response = dict()
+    response['certificates'] = []
+
     try:
         if name:
-            server_certs = [iam.get_server_certificate(ServerCertificateName=name)['ServerCertificate']]
+            response['certificates'].append(connection.get_server_certificate(ServerCertificateName=name))
+        elif path:
+            server_certs = connection.list_server_certificates(PathPrefix=path)['ServerCertificateMetadataList']
         else:
-            server_certs = iam.list_server_certificates()['ServerCertificateMetadataList']
+            server_certs = connection.list_server_certificates()['ServerCertificateMetadataList']
 
-        for server_cert in server_certs:
-            if not name:
-                server_cert = iam.get_server_certificate(ServerCertificateName=server_cert['ServerCertificateName'])['ServerCertificate']
-            cert_md = server_cert['ServerCertificateMetadata']
-            results[cert_md['ServerCertificateName']] = {
-                'certificate_body': server_cert['CertificateBody'],
-                'server_certificate_id': cert_md['ServerCertificateId'],
-                'server_certificate_name': cert_md['ServerCertificateName'],
-                'arn': cert_md['Arn'],
-                'path': cert_md['Path'],
-                'expiration': cert_md['Expiration'].isoformat(),
-                'upload_date': cert_md['UploadDate'].isoformat(),
-            }
+        if not name:
+            for server_cert in server_certs:
+                response['certificates'].append(connection.get_server_certificate(ServerCertificateName=server_cert['ServerCertificateName']))
```

Comment:
@ryansb I agree with @wimnat. The existing module is not inlines with the other _facts module. 

Also please note that the existing module although contains the code to retrieve all server certificates, it is not used since `name` is marked as required. I have now added both `name` and `path` attribute to make this module more complete. 

With regards to the module output, I just tried to make it consistent with the other _facts modules.